index.src.html

```diff
@@ -548,6 +548,34 @@ <h1>Credential Management Level 1</h1>
     to guide the discovery process (by choosing a federated identity provider, BTLE device, etc).
   </div>
 
+  <div algorithm="copy any buffer sources">
+    To <dfn>create a copy of any buffer sources' data</dfn> referenced by a value |V| of
+    [=dictionary=] type, run the following steps given |V|:
+
+    <ol class="algorithm">
+		1. Let |dictionaries| be a list consisting of |V| and all of |V|'s 
+			[=inherited dictionaries=], in order from least to most derived.
+
+		2. For each dictionary |dictionary| in |dictionaries|, in order:
+
+			1. For each dictionary member |member| declared on 
+				 |dictionary|, in lexicographical order:
+
+				1. Let |key| be the [=identifier=] of |member|.
+
+				2. If the dictionary member named |key| is [=present=] in |V|, then:
+
+					1. Let |value| be the value of |member| on |V|.
+
+					2. If |value|'s type is a [=buffer source type=] then 
+						[=get a copy of the bytes held by the buffer source=] and 
+						set |values|'s value to be a reference to the copy of the bytes.
+
+                    issue: do we need to explicitly free the bytes we just orphaned?
```

Comment:
The bytes that belong to `|V|` [`|member|`] aren't orphaned, as they are still rooted to `|dictionary|`. The return could get orphaned, but that's not this algorithm's fault.

Reply:
I was only referring to the buffer source bytes that we copied from and then overwrote the dictionary's references to them  -- do we just assume there is garbage collection?